types/wlr_surface.c

```diff
@@ -698,6 +701,7 @@ static void subsurface_destroy(struct wlr_subsurface *subsurface) {
 	wl_list_remove(&subsurface->surface_destroy.link);
 
 	if (subsurface->parent) {
+		subsurface->parent->pending.committed |= WLR_SURFACE_STATE_SUBSURFACES;
```

Comment:
Subsurfaces are destroyed instantly, their removal isn't double-buffered.

Reply:
Yes, but I'd still like to give the caller a chance to figure out that the subsurface list changed…

Reply:
I think callers should simply listen to new_subsurface/destroy signals, and WLR_SURFACE_STATE_SUBSURFACES should be left as an optimization thing.